### PR TITLE
ci: try to release deb package

### PR DESCRIPTION
**Pull request by PsiACE**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

This is the first step towards releasing the packages with [nfpm](https://nfpm.goreleaser.com/)

### Usage

1. download our release
2. unpack to a path
3. export env: `name` `version` `path`
4. combine `envsubst` and `nfpm`

### example

```
export name="databend"
export version="0.1.0"
export path = "../../../assets"
nfpm pkg --packager deb -f <(envsubst '${nfpm} ${version} ${path}' < nfpm.yaml)
```

### result

```
dpkg -X databend_0.1.0_amd64.deb check
./etc/
./etc/databend/
./etc/databend/databend-meta.toml
./etc/databend/databend-query.toml
./usr/
./usr/local/
./usr/local/bin/
./usr/local/bin/databend-meta
./usr/local/bin/databend-metabench
./usr/local/bin/databend-metactl
./usr/local/bin/databend-query
./usr/local/share/
./usr/local/share/doc/
./usr/local/share/doc/databend/
./usr/local/share/doc/databend/readme.txt
```

### CODE REVIEW

File: .github/actions/publish_binary/nfpm.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+description: |
+  Databend is a powerful cloud data warehouse. Built for elasticity and efficiency.
+  Free and open. Also available in the cloud: https://app.databend.com
+vendor: "Databend"
+homepage: "https://databend.rs"
+license: "Apache-2.0"
+contents:
+  # Binaries
+  - src: ${path}/bin/databend-query
+    dst: /usr/local/bin/databend-query
```
[the remark this answers is not in the document]
Comment (PsiACE) [reply .github/actions/publish_binary/nfpm.yaml:17]: OK, I'll change it

File: .github/actions/publish_binary/nfpm.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+arch: "amd64"
+platform: "linux"
+version: "${version}"
+section: "database"
+priority: "extra"
+maintainer: "Databend Authors <opensource@datafuselabs.com>"
+description: |
+  Databend is a powerful cloud data warehouse. Built for elasticity and efficiency.
+  Free and open. Also available in the cloud: https://app.databend.com
+vendor: "Databend"
```
Comment (Xuanwo) [inline .github/actions/publish_binary/nfpm.yaml:11]: Datafuse Labs

File: .github/actions/publish_binary/nfpm.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  - src: ${path}/configs/databend-query.toml
+    dst: /etc/databend/databend-query.toml
+    type: config
+  - src: ${path}/configs/databend-meta.toml
+    dst: /etc/databend/databend-meta.toml
+    type: config
+
+  # Docs
+  - src: ${path}/readme.txt
+    dst: /usr/local/share/doc/databend/readme.txt
```
Comment (everpcpc) [inline .github/actions/publish_binary/nfpm.yaml:35]:
```suggestion
    dst: /usr/share/doc/databend/readme.txt
```

File: .github/workflows/databend-release.yml [hunk 4]
```diff
@@ -346,6 +346,52 @@ jobs:
...
+  release_packages:
+    name: release packages
+    runs-on: ubuntu-latest
+    needs: [create_release, publish_linux]
+    strategy:
+      matrix:
+        arch:
+          - x86_64
+        platform:
+          - musl
```
Comment (everpcpc) [inline .github/workflows/databend-release.yml:358]: maybe we should release gnu build?

**Review by everpcpc — APPROVED** [review]

File: .github/workflows/databend-release.yml (hunk 4 above)
Comment (PsiACE) [reply .github/workflows/databend-release.yml:358]: emmm, let's start by confirming that this workflow is correct and that switching targets is easy.

**Pull request merged**